### ISSUE TITLE
AngularJS issue

### ISSUE THREAD

**Issue by edclement** [WickyNilliams/headroom.js#204]
The following code needs to get reversed in the latest 0.8 release of the `angular.headroom.js` file:
```
angular.forEach(Headroom.options, function(value, key) {
  options[key] = scope[key] || Headroom.options[key];
});
if (options.scroller) {
  options.scroller = document.querySelector(options.scroller);
}
```
**TO**
```
if (options.scroller) {
  options.scroller = document.querySelector(options.scroller);
}
angular.forEach(Headroom.options, function(value, key) {
  options[key] = scope[key] || Headroom.options[key];
});
```

It appears that the `Headroom.options.scroller` is now defined so the `if` block below it always gets invoked. So if you don't pass a `scroller` option in the headroom directive, the browser `window` object gets passed to the `document.querySelector` call, thus throwing an error. It basically makes it impossible to default to the `window` object as the scroll trigger.

**Comment by toddmotto** [comment]
Interesting, I did fix this in this PR by scoping out the `[object Window]`: https://github.com/WickyNilliams/headroom.js/pull/202

**Comment by WickyNilliams** [comment]
This actually seems like a cleaner fix than sniffing the window @toddmotto. Perhaps adopt this in your PR?

**Comment by edclement** [comment]
Yeah, I just left a comment on https://github.com/WickyNilliams/headroom.js/pull/202/files

**Comment by toddmotto** [comment]
Agreed, done: https://github.com/WickyNilliams/headroom.js/pull/202/commits/e24251c83ea345028cef19efda2bd5a80fb28f5d

**Comment by edclement** [comment]
Thanks for the quick follow up :+1:

**Comment by WickyNilliams** [comment]
Released in 0.9.1
